lib/soapy_cake/client.rb

```diff
@@ -1,18 +1,15 @@
+# frozen_string_literal: true
 module SoapyCake
   class Client
     HEADERS = { 'Content-Type' => 'application/soap+xml;charset=UTF-8' }.freeze
 
     def initialize(opts = {})
-      @domain = opts.fetch(:domain, ENV['CAKE_DOMAIN']) || fail(Error, 'Cake domain missing')
-      @api_key = opts.fetch(:api_key, ENV['CAKE_API_KEY']) || fail(Error, 'Cake API key missing')
-      @retry_count = opts.fetch(:retry_count, ENV['CAKE_RETRY_COUNT']) || 4
-
-      @time_converter = TimeConverter.new(
-        opts.fetch(:time_zone, ENV['CAKE_TIME_ZONE']),
-        opts.fetch(:time_offset, ENV['CAKE_TIME_OFFSET'])
-      )
-
       @opts = opts
+      @domain = fetch_opt(:domain) || raise(Error, 'Cake domain missing')
+      @api_key = fetch_opt(:api_key) || raise(Error, 'Cake API key missing')
+      @retry_count = fetch_opt(:retry_count, 4)
+      @write_enabled = ['yes', true].include?(fetch_opt(:write_enabled))
```

Comment:
I changed this to a bit more explicit (so e.g. `CAKE_WRITE_ENABLED=0` doesn't count).  Went with `yes`, as we're already using that pattern in some of the apps.